### ISSUE TITLE
I have a question about a proxy support

### ISSUE THREAD

**Issue by c-rei** [box/box-python-sdk-gen#318]
I've read the docs about to use Proxy for API calls just like below: 

***** docs *****
Use Proxy for API calls
In order to use a proxy for API calls, calling the client.with_proxy(proxyConfig) method creates a new client, leaving the original client unmodified, with the username and password being optional.

new_client = client.with_proxy(
    ProxyConfig(url="http://proxy.com", username="username", password="password")
)
***** ended *****

But, I don't understand that in order to get a original client behind proxy, I have to use API call for oauth2 authentication, and this time I don't have a proxy config.

So I want to known that I can get a oauth2 authentication without proxy config? The oauth2 code just like your docs:

***** docs *****
auth = BoxOAuth(
    OAuthConfig(client_id="YOUR_CLIENT_ID", client_secret="YOUR_CLIENT_SECRET")
)
auth_url = auth.get_authorize_url()

auth.get_tokens_authorization_code_grant("YOUR_ACCESS_CODE")   <-- get access code at my redirect URL.
client = BoxClient(auth)        <- this is a original client. then I can use it to get a client for proxy call?
***** ended *****

**Comment by c-rei** [comment]
And I've tried do a proxy connection with sample code:

*******************
new_client = client.with_proxy(
    ProxyConfig(url="http://proxy.com", username="username", password="password")
)
items_in_root_folder = [
    item.name for item in new_client.folders.get_folder_items(folder_id="0").entries
]
*******************
I got an error:

Message: HTTPSConnectionPool(host='api.box.com', port=443): Max retries exceeded with url: /2.0/folders/0/items (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000026CA93F99C0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')))

Note: I can use the same proxy config with old box-python-sdk.

**Comment by lukaszsocha2** [comment]
Hi @c-rei,
so first of all if you want to use proxy to get OAuth token you can use code:
```
auth = BoxOAuth(
OAuthConfig(client_id="YOUR_CLIENT_ID", client_secret="YOUR_CLIENT_SECRET")
)
auth_url = auth.get_authorize_url()

#  After callback
network_session_with_proxy = NetworkSession().with_proxy(config=ProxyConfig(url="http://proxy.com/", username="username", password="password"))
auth.get_tokens_authorization_code_grant("YOUR_ACCESS_CODE", network_session=network_session_with_proxy)
client = BoxClient(auth, network_session=network_session_with_proxy)
```
if you don't want to use proxy to perform `get_tokens_authorization_code_grant` call just skip `network_session` parameter.
I'll try to debug you issue from the second message and update you soon.
Best, 
@lukaszsocha2

**Comment by lukaszsocha2** [comment]
Hi @c-rei,
can you print
```
print(client.network_session.requests_session.proxies)
```
using your client with proxy set to verify if the proxy setting are as expected.
Please let us know about results,
Best, 
@lukaszsocha2

**Comment by c-rei** [comment]
Hi, @lukaszsocha2 
I'll give your results with print.

At old sdk:
print(client.session._prepare_proxy())
{'http': 'http://192.168.200.12:3128', 'https': 'http://192.168.200.12:3128'}

And new sdk:
print(new_client.network_session.requests_session.proxies)
{'http': 'http://None192.168.200.12:3128', 'https': 'http://None192.168.200.12:3128'}

So, I think you have a problem in this code:
![image](https://github.com/user-attachments/assets/bb0f2f05-941c-435e-ae8b-813d31bd6ced)

**Comment by c-rei** [comment]
I've tried that:
f'aaa{None}, and the result is "aaaNone".

**Comment by lukaszsocha2** [comment]
Hi,
thanks for the help. We will fix this and make a new release today.
Best,
@lukaszsocha2

**Comment by lukaszsocha2** [comment]
Hi @c-rei,
the fix have been merged, so the issue was closed. Let us know if the fix worked for you. If you still have any problems with proxy feel free to reopen this issue or create a new one. A new release with the fix will be done today or tomorrow.
Best,
@lukaszsocha2

**Comment by lukaszsocha2** [comment]
Hi @c-rei ,
the fix is now available in the new v1.5.1 release.

**Comment by c-rei** [comment]
Hi, @lukaszsocha2 
The fix worked for me.
Thanks for your update.
